client/reader/tag-stream/header.jsx

```diff
@@ -5,77 +5,106 @@ import React from 'react';
 
-const TagStreamHeader = (
-		{ tag, tagImage, isPlaceholder, showFollow, following, onFollowToggle, translate, hasBackButton }
-	) => {
-	const classes = classnames( {
-		'tag-stream__header': true,
-		'is-placeholder': isPlaceholder,
-		'has-back-button': hasBackButton
-	} );
-	const imageStyle = {};
+class TagStreamHeader extends React.Component {
 
-	if ( tagImage ) {
-		const imageUrl = resizeImageUrl( tagImage.url, { w: TAG_HEADER_WIDTH } );
-		const safeCssUrl = cssSafeUrl( imageUrl );
-		imageStyle.backgroundImage = 'url(//' + safeCssUrl + ')';
+	static propTypes = {
+		isPlaceholder: React.PropTypes.bool,
+		tag: React.PropTypes.string,
+		showFollow: React.PropTypes.bool,
+		following: React.PropTypes.bool,
+		onFollowToggle: React.PropTypes.func,
+		tagImages: React.PropTypes.array,
+		hasBackButton: React.PropTypes.bool
+	};
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			tag: props.tag,
+			tagImages: props.tagImages
+		};
+		this.pickNewTagImage( props );
 	}
 
-	return (
-		<div className={ classes }>
-			<QueryReaderTagImages tag={ tag } />
-			{ showFollow &&
-				<div className="tag-stream__header-follow">
-					<FollowButton
-						followLabel={ translate( 'Follow Tag' ) }
-						followingLabel={ translate( 'Following Tag' ) }
-						iconSize={ 24 }
-						following={ following }
-						onFollowToggle={ onFollowToggle } />
-				</div>
-			}
+	componentWillReceiveProps( nextProps ) {
+		this.pickNewTagImage( nextProps );
+	}
+
+	pickNewTagImage = ( props = this.props ) => {
+		if ( this.state && this.state.tagImages !== props.tagImages ) {
+			this.state = {
+				tag: props.tag,
+				tagImages: props.tagImages,
+				chosenTagImage: sample( props.tagImages )
+			};
+		}
+	}
 
-			<div className="tag-stream__header-image" style={ imageStyle }>
-				<h1 className="tag-stream__header-image-title">
-					<Gridicon icon="tag" size={ 24 } />{ tag }
-				</h1>
-				{ tagImage &&
-					<div className="tag-stream__header-image-byline">
-						<span className="tag-stream__header-image-byline-label">{ translate( 'Photo by' ) }</span> <a href={ tagImage.blog_url } className="tag-stream__header-image-byline-link" rel="author external">{ decodeEntities( tagImage.author ) }</a>, <a href={ tagImage.blog_url } className="tag-stream__header-image-byline-link" rel="external">{ decodeEntities( tagImage.blog_title ) }</a>
+	render() {
+		const { tag, isPlaceholder, showFollow, following, onFollowToggle, translate, hasBackButton } = this.props;
+		const classes = classnames( {
+			'tag-stream__header': true,
+			'is-placeholder': isPlaceholder,
+			'has-back-button': hasBackButton
+		} );
+		const imageStyle = {};
+		const tagImage = this.state.chosenTagImage;
+
+		if ( tagImage ) {
+			const imageUrl = resizeImageUrl( 'https://' + tagImage.url, { resize: `${ TAG_HEADER_WIDTH },${ TAG_HEADER_HEIGHT }` } );
+			const safeCssUrl = cssSafeUrl( imageUrl );
+			imageStyle.backgroundImage = 'url(' + safeCssUrl + ')';
```

Comment:
maybe we should make up a `<BackgroundImage/>` component that does all this for us. We seem to keep doing it.